src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs

```diff
@@ -149,217 +130,170 @@ public void SetCapability(CompletionCapability capability, ClientCapabilities cl
-                            Character = completionRange.End.Column - 1
-                        }
+                        Kind = CompletionItemKind.Folder,
+                        InsertTextFormat = InsertTextFormat.Snippet,
+                        TextEdit = textEdit with { NewText = snippet }
                     }
-                }
+                    : item with { Kind = CompletionItemKind.Folder },
+                CompletionResultType.Property => item with { Kind = CompletionItemKind.Property },
+                CompletionResultType.Method => item with { Kind = CompletionItemKind.Method },
+                CompletionResultType.ParameterName => ExtractTypeFromToolTip(detail, out string type)
+                    ? item with { Kind = CompletionItemKind.Variable, Detail = type }
+                    // The comparison operators (-eq, -not, -gt, etc) unfortunately come across as
+                    // ParameterName types but they don't have a type associated to them, so we can
+                    // deduce it is an operator.
+                    : item with { Kind = CompletionItemKind.Operator },
+                CompletionResultType.ParameterValue => item with { Kind = CompletionItemKind.Value },
+                CompletionResultType.Variable => ExtractTypeFromToolTip(detail, out string type)
+                    ? item with { Kind = CompletionItemKind.Variable, Detail = type }
+                    : item with { Kind = CompletionItemKind.Variable },
+                CompletionResultType.Namespace => item with { Kind = CompletionItemKind.Module },
+                CompletionResultType.Type => detail.StartsWith("Class ", StringComparison.CurrentCulture)
+                    // Custom classes come through as types but the PowerShell completion tooltip
+                    // will start with "Class ", so we can more accurately display its icon.
+                    ? item with { Kind = CompletionItemKind.Class }
+                    : item with { Kind = CompletionItemKind.TypeParameter },
+                CompletionResultType.Keyword or CompletionResultType.DynamicKeyword =>
+                    item with { Kind = CompletionItemKind.Keyword },
+                _ => throw new ArgumentOutOfRangeException(nameof(result))
             };
         }
 
-        private static CompletionItemKind MapCompletionKind(CompletionType completionType)
+        /// <summary>
+        /// Look for type encoded in the tooltip for parameters and variables. Display PowerShell
+        /// type names in [] to be consistent with PowerShell syntax and how the debugger displays
+        /// type names.
+        /// </summary>
+        /// <param name="toolTipText"></param>
+        /// <param name="type"></param>
+        /// <returns>Whether or not the type was found.</returns>
+        private static bool ExtractTypeFromToolTip(string toolTipText, out string type)
         {
-            switch (completionType)
+            Regex _typeRegex = new(@"^(\[.+\])");
```

Comment:
```suggestion
            Regex typeRegex = new(@"^(\[.+\])");
```

For locals skip `_` (I know it's to mirror the real field but still, better to just reference the new local)